libcontainer/cgroups/fs2/memory.go

```diff
@@ -124,6 +124,9 @@ func statMemory(dirPath string, stats *cgroups.Stats) error {
 	if swapUsage.Limit != math.MaxUint64 {
 		swapUsage.Limit += memoryUsage.Limit
 	}
+	// The `MaxUsage` of mem+swap cannot simply combine mem with
+	// swap. So set it to 0 for v1 compatibility.
+	swapUsage.MaxUsage = 0
```

Comment:
😢 This is also I want to say. So we read the value from `memory.swap.peak`, and then clear it?
But I don't know a way to resolve it.

Reply:
Cgroup v2 breaks some compatibilities with v1, and `memsw` is one of them. Now the code is aligned with the old, to maintain the compatibility. But maybe we don't need this compatibility when showing cgroup v2 stat? I don't sure about it.

Reply:
@opencontainers/runc-maintainers How about add an new stats named `SwapOnlyUsage` in:
https://github.com/opencontainers/runc/blob/8fa126ac7513ba310ab61fa4f5a7de53ef736513/libcontainer/cgroups/stats.go#L87-L106

Reply:
I think it's ok, @AkihiroSuda wdyt

Reply:
SGTM

Reply:
If we accept `SwapOnlyUsage`, I think this stats should be filled correctly in cgroup v1?

Reply:
In cgroups v1 mode, the swap usage is always combined with memory usage. So it is the same problem as before. Now the `SwapUsage` will be used in v1, and `SwapOnlyUsage` will be used in v2. Though it's a quite weird thing, it's true for the kernel.

Reply:
@lifubang Sorry for the slow response. I've implemented the filling in cgroups v1 mode. Pls check.

Reply:
if we want compatibility with cgroupv1, why don't we do `swapUsage.MaxUsage += memoryUsage.MaxUsage`? then callers can remove the memoryUsage.MaxUsage to find the swap-only stats--similar to the other swap fields from v1

Reply:
In my opinion, `max(memory + swap) != max(memory) + max(swap)`. The time of hitting max memory usage may be different from the time hitting max swap usage. So the result will be more than the actual memsw usage.